vortex-scalar/src/value.rs

```diff
@@ -28,6 +29,26 @@ impl ScalarValue {
         matches!(self, Self::Null)
     }
 
+    pub fn is_instance_of(&self, dtype: &DType) -> bool {
+        match (self, dtype) {
+            (ScalarValue::Bool(_), DType::Bool(_)) => true,
+            (ScalarValue::Primitive(pvalue), DType::Primitive(ptype, _)) => {
+                pvalue.is_instance_of(ptype)
+            }
+            (ScalarValue::Buffer(_), DType::Binary(_)) => true,
+            (ScalarValue::BufferString(_), DType::Utf8(_)) => true,
+            (ScalarValue::List(values), DType::List(dtype, _)) => {
+                values.iter().all(|v| v.is_instance_of(dtype))
+            }
+            (ScalarValue::List(values), DType::Struct(structdt, _)) => values
```

Comment:
typo, this should be ScalarValue::Struct on left

Reply:
Rob & Nick argue against adding ScalarValue::Struct b/c field name duplication: https://spiraldb.slack.com/archives/C07BV3GKAJ2/p1727813531212969